book/website/project-design/missing-data/missing-data-visualising-missingness.md

```diff
@@ -0,0 +1,256 @@
+### 1. Nullity Matrix 
+```python
+msno.matrix(df_fictional_dataset)
+```
+```{figure} ../../figures/msno_matrix.png
+---
+height: 400px
+name: msno-matrix
+alt: 
+---
+Nullity matrix of our fictional dataset, produced via the missingno python package. 
+```
+Data entries with missing data are indicated by white, while all complete entries are shaded a dark grey. At the right of the plot there is a sparkline which summarizes data completeness by row. In this instance the 3rd row has the most missing data (only 2 complete entires), while the 4th and 5th rows are the most complete (no missing entries).
+
+
+(pd-missing-data-visualising-missingness-python2)=
+### 2. Nullity by Column
+```python
+msno.bar(df_fictional_dataset)
+```
+```{figure} ../../figures/msno_bar.png
+---
+height: 400px
+name: msno-bar
+alt: 
+---
+Barplot of the nullity per column of our fictional dataset, produced via the missingno python package. 
+```
+This is a simplification of the first visualization method; a simple bar plot showing the number of complete values per variable. The absolute number of values are shown on the top, while the left y-axis indicates the percentage of values that are not missing. 
+
+(pd-missing-data-visualising-missingness-python3)=
+### 3. Nullity Correlation
+```python
+msno.heatmap(df_fictional_dataset)
+```
+```{figure} ../../figures/msno_heatmap.png
+---
+height: 400px
+name: msno-heatmap
+alt: 
+---
+Pairwise nullity correlation heatmap of our fictional dataset, produced via the missingno python package.
+```
+ Visualizing what data is missing may not be enough to draw any inferences and understand the mechanism behind the missingness. A nullity correlation plot is one way to look into this. In this instance, we see that systolic blood pressure is always missing when diastolic blood pressure is also missing. Additionally, the cognitive score has a 50% chance of being missing if the blood pressure readings are missing. As the motor score and age don't have any missing values these are excluded. 
+
+
+(pd-missing-data-visualising-missingness-python4)=
+### 4. Beyond Pairwise Nullity Correlations, via a hierarchical clustering algorithm  
+```python
+msno.dendrogram(df_fictional_dataset)
+```
+```{figure} ../../figures/msno_dendrogram.png
+---
+height: 400px
+name: msno-dendrogram
+alt: 
+---
+Nullity correlation dendrogram of our fictional dataset, produced via the missingno python package.
+```
+
+
+In the dendrogram, the missingness of variables that are linked at a distance of zero, are fully correlated with each other. Thus, we see the Motor Score and Age being connected at zero, and separately the two blood pressure measures being connected. Variables the would branch close to zero, predict eachothers missingness well, but not perfectly. For instance, cognitive score branches from the blood pressure variables at around 1.30. This is because 2 out of 3 missing cognitive score values, also have missing blood pressure measurements. 
```

Comment:
Variables the would branch close to zero, predict eachothers missingness well, but not perfectly.

This sentence doesn't quite make sense?